### PR TITLE
feat: adds status to the github-pull-requests-board

### PR DESCRIPTION
**Pull request by Josh-Uvi**
This relates to #20546
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/chilled-dolphins-tap.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-github-pull-requests-board': patch
+---
+
+The cardheader component in the github-pull-requests-board plugin now requires that a `status` is passed to the component.
```
Comment (benjdlambert) [inline .changeset/chilled-dolphins-tap.md:5]:
```suggestion
The `CardHeader` component in the `github-pull-requests-board` plugin will show the status for the PR
```

File: plugins/github-pull-requests-board/src/api/useGetPullRequestDetails.ts [hunk 2]
```diff
@@ -57,6 +57,15 @@ export const useGetPullRequestDetails = () => {
                     state
                   }
                 }
+                commits(last: 1) {
+                  nodes {
+                    commit {
+                      statusCheckRollup {
+                        state
```
Comment (benjdlambert) [inline plugins/github-pull-requests-board/src/api/useGetPullRequestDetails.ts:64]: Does this always return a status? Can this be undefined at any point?

File: plugins/github-pull-requests-board/src/components/Card/Card.tsx [hunk 3]
```diff
@@ -29,6 +29,7 @@ type Props = {
   isDraft: boolean;
   repositoryIsArchived: boolean;
   labels?: Label[];
+  status: Status;
```
Comment (benjdlambert) [inline plugins/github-pull-requests-board/src/components/Card/Card.tsx:32]: Wonder if it makes sense to make `status` optional, and only display it if it's passed for all the components?

File: plugins/github-pull-requests-board/src/components/Card/CardHeader.tsx [hunk 4]
```diff
@@ -86,6 +88,12 @@ const CardHeader: FunctionComponent<Props> = (props: Props) => {
           </Typography>
         )}
       </Box>
+      <Box display="flex" alignItems="center" flexWrap="wrap" paddingTop={1}>
```
Comment (benjdlambert) [inline plugins/github-pull-requests-board/src/components/Card/CardHeader.tsx:91]: Basically same as above

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Thanks for this @Josh-Uvi!

Just a few questions!

File: plugins/github-pull-requests-board/src/api/useGetPullRequestDetails.ts (hunk 2 above)
Comment (Josh-Uvi) [reply plugins/github-pull-requests-board/src/api/useGetPullRequestDetails.ts:64]: Each commit always returns a status, which can be one of the following: ERROR, EXPECTED, FAILURE, PENDING, or SUCCESS.

File: plugins/github-pull-requests-board/src/components/Card/Card.tsx (hunk 3 above)
Comment (Josh-Uvi) [reply plugins/github-pull-requests-board/src/components/Card/Card.tsx:32]: I think that's a good idea, I'll make the necessary changes

Comment (Josh-Uvi) [reply plugins/github-pull-requests-board/src/components/Card/Card.tsx:32]: As mentioned in #20546 - `being able to see which ones are already passing their checks without having to click into each one makes prioritizing them for review much faster.`
@benjdlambert I believe it's better to show all statuses regardless of whether it's passed.

File: plugins/github-pull-requests-board/src/components/Card/CardHeader.tsx [hunk 5]
```diff
@@ -91,7 +91,7 @@ const CardHeader: FunctionComponent<Props> = (props: Props) => {
       <Box display="flex" alignItems="center" flexWrap="wrap" paddingTop={1}>
         <Typography variant="body2" component="p">
           Commit Status:{' '}
-          <strong>{status.commit.statusCheckRollup.state}</strong>
+          <strong>{status?.commit.statusCheckRollup.state}</strong>
```
Comment (benjdlambert) [inline plugins/github-pull-requests-board/src/components/Card/CardHeader.tsx:94]: Maybe?

```suggestion
          <strong>{status?.commit.statusCheckRollup.state ?? 'unknown'}</strong>
```

Comment (benjdlambert) [reply plugins/github-pull-requests-board/src/components/Card/CardHeader.tsx:94]: Or just hide if `status` is undefined?

Comment (Josh-Uvi) [reply plugins/github-pull-requests-board/src/components/Card/CardHeader.tsx]: Done - hide status if undefined

**Review by benjdlambert — APPROVED** [review]
Nice :pray: thanks!

**Pull request merged**